doc/cprover-manual/contracts-assigns.md

```diff
@@ -1,75 +1,368 @@
 [CPROVER Manual TOC](../../)
 
-# Assigns Clause
+# Assigns Clauses
 
-## In Function Contracts
+An _assigns clause_ lets the user to specify which memory locations may be
+assigned to by a function or a loop.
 
-### Syntax
+For a function contract, if no assigns clause is provided, the default is the
+empty set.
+
+For a loop contract, if no assigns clause is provided, CBMC attempts to infer
+the set of locations assigned by the loop from the loop body,
+and then checks that the inferred set is correct (as if specified by the user).
```

Comment:
One of our goals is to make contracts standalone and independent of a particular function, in which case there is no programs to infer a clause from.

One way to figure out the side effects of a particular function however could be to check it under an empty assigns clause and to turn all reported violations into an assigns clause.

For loops it is different because they necessarily live in a function and just happen to assign whatever they assign.